okhttp-tests/src/test/java/okhttp3/HttpUrlTest.java

```diff
@@ -1328,24 +1325,26 @@
   }
 
   @Test public void fromJavaNetUrl_checked() throws Exception {
-    HttpUrl httpUrl = HttpUrl.getChecked("http://username:password@host/path?query#fragment");
-    assertEquals("http://username:password@host/path?query#fragment", httpUrl.toString());
+    String namePass = "us.er-name:p@a!s s#w$o^r&d*";
+    String encodedNamePass = URLEncoder.encode(namePass, "UTF-8");
+    String url = "http://" + encodedNamePass + "@host/path?query#fragment";
+    HttpUrl httpUrl = HttpUrl.getChecked(url);
+    assertEquals(url, httpUrl.toString());
   }
 
-  @Test public void fromJavaNetUrlUnsupportedScheme_checked() throws Exception {
-    try {
-      HttpUrl.getChecked("mailto:user@example.com");
-      fail();
-    } catch (MalformedURLException e) {
-    }
+  @Test(expected = MalformedURLException.class)
+  public void fromJavaNetUrlUnsupportedScheme_checked() throws Exception {
+    HttpUrl.getChecked("mailto:user@example.com");
   }
 
-  @Test public void fromJavaNetUrlBadHost_checked() throws Exception {
-    try {
-      HttpUrl.getChecked("http://hostw ithspace/");
-      fail();
-    } catch (UnknownHostException expected) {
-    }
+  @Test(expected = UnknownHostException.class)
+  public void fromJavaNetUrlBadHost_checked() throws Exception {
+    HttpUrl.getChecked("http://hostw ithspace/");
+  }
+
```

Comment:
I don't think this test is necessary. We don't maintain binary compatibility through unit tests.